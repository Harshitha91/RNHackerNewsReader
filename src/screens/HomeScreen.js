import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { ListItem } from '../components';
import { View, StyleSheet, FlatList } from 'react-native';
import { API } from '../constants';

export default function HomeScreen({ navigation }) {
    const [listsData, setListsData] = useState([]);
    const [isRefreshing, setRefreshing] = useState(false);

    useEffect(() => {
        fetchData();
    }, []);

    const fetchData = async () => {
        const response = await axios.get(API + 'topstories.json');
        setListsData(response?.data ?? []);
        setRefreshing(false);
    };

    const onRefresh = () => {
        setRefreshing(true);
        fetchData();
    };

    return (
        <View style={styles.container}>
            <FlatList
                testID={'flatlist'}
                style={styles.list}
                data={listsData}
                horizontal={false}
                keyExtractor={item => item.toString()}
                ItemSeparatorComponent={() => {
                    return (
                        <View testID={'separator'} style={styles.separator} />
                    );
                }}
                refreshing={isRefreshing}
                onRefresh={onRefresh}
                renderItem={({ item }) => {
                    return (
                        <ListItem
                            testID={'list-item'}
                            navigation={navigation}
                            id={item.toString()}
                        />
                    );
                }}
            />
        </View>
    );
}
const styles = StyleSheet.create({
    container: {
        backgroundColor: 'white',
        flex: 1,
    },
    list: {
        backgroundColor: 'transparent',
        height: '100%',
    },
    separator: {
        backgroundColor: '#999999',
        left: 10,
        height: 2,
    },
});
