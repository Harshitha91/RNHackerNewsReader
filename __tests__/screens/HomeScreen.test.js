import * as React from 'react';
import HomeScreen from '../../src/screens/HomeScreen';
import { render, cleanup } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

it(`renders correctly`, async () => {
    const tree = renderer.create(<HomeScreen />);
    expect(tree).toMatchSnapshot();
});

describe('Test Home Screen', () => {
    it('should contain flatlist', () => {
        const component = <HomeScreen />;

        const { getByTestId } = render(component);
        const flatlist = getByTestId('flatlist');

        expect(flatlist).toBeTruthy();
    });

    it('should contain list item seperator', () => {
        const component = <HomeScreen />;

        const { findByTestId } = render(component);
        const separator = findByTestId('separator');

        expect(separator).toBeTruthy();
    });
});
