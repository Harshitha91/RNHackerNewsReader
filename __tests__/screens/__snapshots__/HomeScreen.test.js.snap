// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "transparent",
        "height": "100%",
      }
    }
    testID="flatlist"
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;
